### PR TITLE
feat(auth/basic): extend auth basic command

### PR DESCRIPTION
**Pull request by alacuku**
Username and password can be set using cli flags, env variables and stdin.

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

> /kind flaky-test

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area library

/area cli

/area tests

 /area examples

**What this PR does / why we need it**:

This PR extends the `falcoctl registry auth basic` command by allowing `username` and `password` to be set using the cli flags, env variables and stdin other than the interactive prompt.

See examples:

```bash
Example - Log in with username and password from command line flags:
	falcoctl registry auth basic -u username -p password localhost:5000
Example - Login with username and password from env variables:
	FALCOCTL_REGISTRY_AUTH_BASIC_USERNAME=username FALCOCTL_REGISTRY_AUTH_BASIC_PASSWORD=password falcoctl registry auth basic localhost:5000
Example - Login with username and password from stdin:
	falcoctl registry auth basic -u username --password-stdin localhost:5000
Example - Login with username and password in an interactive prompt:
	falcoctl registry auth basic localhost:5000
```

**Which issue(s) this PR fixes**:

Fixes #
#134 

**Special notes for your reviewer**:

### CODE REVIEW

File: cmd/registry/auth/basic/basic.go [hunk 1]
```diff
@@ -43,22 +52,52 @@ func NewBasicCmd(ctx context.Context, opt *options.Common) *cobra.Command {
...
+
+Example - Login with username and password in an interactive prompt:
+	falcoctl registry auth basic localhost:5000
+`,
+		Args: cobra.ExactArgs(1),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			o.username = viper.GetString("registry.auth.basic.username")
+			o.password = viper.GetString("registry.auth.basic.password")
+			o.passwordFromStdin = viper.GetBool("registry.auth.basic.password_stdin")
+
```
Comment (FedeDP) [inline cmd/registry/auth/basic/basic.go:74]: question, what if we also support automatically fallbacking at interactive prompt for password if only `-u` is passed (ie: if password is empty but username is not)?

Comment (FedeDP) [reply cmd/registry/auth/basic/basic.go:74]: And we deprecated `--password-stdin`

Comment (alacuku) [reply cmd/registry/auth/basic/basic.go:74]: > question, what if we also support automatically fallbacking at interactive prompt for password if only `-u` is passed (ie: if password is empty but username is not)?

We already do that. If the password is not set using the -p, --password, or the env variable it falls back at the interactive prompt.

Comment (alacuku) [reply cmd/registry/auth/basic/basic.go:74]: > And we deprecate `--password-stdin`

The `--password-stdin` is used when piping the password from another command. For example:
```bash
echo $PASSWORD | ./falcoctl registry auth basic -u username localhost:5000 --password-stdin

```

Comment (FedeDP) [reply cmd/registry/auth/basic/basic.go:74]: Ah makes sense, thanks!

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**